src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IepLevelService.kt

```diff
@@ -16,4 +17,56 @@ class IepLevelService(
       .toList()
       .sortedWith(compareBy(IepLevel::sequence))
   }
+
+  /**
+   * Because not all levels are enabled at all prisons, and we want to maintain a
+   * person’s level when they are transferred, we need to find the "nearest highest" level
+   * according to HMPPS policy. Falls back to prison's default level when there's nothing else to do.
+   */
+  suspend fun findNearestHighestLevel(prisonId: String, levelCode: String): String {
+    var maybeDefaultLevelCode: String? = null
+    val levelCodesAvailableInPrison = getIepLevelsForPrison(prisonId, true)
```

Comment:
i don't know whether to `useClientCredentials` here :/

Reply:
also i can't figure out the role needed to call `prisonApiService.getIepLevels()` / GET "/api/reference-domains/domains/{domain}/codes/{code}". possibly none??

Reply:
> i don't know whether to useClientCredentials here or not :/

We need to because the request originates from `processOffenderEvent` which is invoked when we receive an SQS message, which unlike HTTP endpoints being called by other APIs, does not have client credentials in the header so we need to use our own when making subsequent HTTP calls

And yep looking at https://github.com/ministryofjustice/prison-api/blob/main/src/main/java/uk/gov/justice/hmpps/prison/api/resource/ReferenceDomainResource.java - looks like no role is required for that endpoint